src/classification/source/classification-source.repository.ts

```diff
@@ -0,0 +1,41 @@
+import { ClassificationSourceModel } from './classification-source.model';
+
+export class ClassificationSourceRepository {
+    static getSearchedUserSources(userId: string, searchFilter: string = '', isSysAdmin: boolean = false) {
+        if (!!isSysAdmin) {
+            return ClassificationSourceModel.aggregate([
+                { $match: { name: { $regex: searchFilter, $options: 'i' } } },
+                { $project: { id: '$_id', _id: 0, name: 1 } },
+                { $sort: { name: 1 } },
+            ]);
+        }
+
+        return ClassificationSourceModel.aggregate([
```

Comment:
This could be changed to:

```javascript
return ClassificationSourceModel.aggregate([
 { $match: { name: { $regex: searchFilter, $options: 'i' } } },
            isSysAdmin 
                ? ...[]
                : ...[{
                $lookup: {
                    from: 'userclassifications',
                    let: { cId: '$classificationId', usr: '$user', lyr: '$layer' },
                    pipeline: [
                        {
                            $match: {
                                $expr: {
                                    $and: [
                                        { $eq: ['$user', userId] },
                                        { $eq: ['$$cId', '$classificationId'] },
                                        { $lte: ['$$lyr', '$layer'] },
                                    ],
                                },
                            },
                        },
                    ],
                    as: 'userClassifications',
                },
            },
            { $unwind: '$userClassifications' },
            { $match: { userClassifications: { $exists: true } } }
            ],
            { $project: { id: '$_id', _id: 0, name: 1 } },
            { $sort: { name: 1 } },
        ]);
```

And then no need for the upper if statement (`isSysAdmin`).